massgov/mayflower
# DP-9200 More link context in action finder

Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/dev/docs/change-log-instructions.md)


## Description

This change is related to Drupal change. Mayflower change needs to be released before the Drupal's.

No visual change.
- Add context info for "more" link in decorative link.
- Set a condition to add **title** attribute only if its unique value is available, and not duplicate the link label. 
- Add the new variable for the context info to the service page json(patternlab/styleguide/source/_patterns/05-pages/service.json).

## Related Issue / Ticket

- [DP-9200 \[a11y\] Missing context on more links](https://jira.mass.gov/browse/DP-9200)
- [DP-9200 \[a11y\] More link context in action finder #2962](https://github.com/massgov/mass/pull/2962)

## Steps to Test


1. Go to the service page: http://localhost:3000/?p=pages-service
2. Navigate to the **See all** links of the **What would you like to do?** and the **What you need to know**.
3. Check the mark up of the the **See all** links. Find the visually hidden text of the title of the action finder. 
```
<div class="ma__action-finder__see-all">
  <div class="ma__action-finder__see-all-container">
    <span class="ma__decorative-link button-link">
      <a href="http://www.google.com" class="js-clickable-link" "="">
        See all 15
        <span class="ma__visually-hidden">  of What would you like to do?</span>
        &nbsp;<svg aria-hidden="true"><use xlink:href="#27404a1a788cb213d119ddcafde3834c.1545"></use></svg><svg xmlns="http://www.w3.org/2000/svg" style="display: none"><symbol xmlns="http://www.w3.org/2000/svg" aria-hidden="true" version="1.1" viewBox="0 0 16 18" id="27404a1a788cb213d119ddcafde3834c.1545"><path d="M983.721 1887.28L983.721 1887.28L986.423 1890L986.423 1890L986.423 1890L983.721 1892.72L983.721 1892.72L978.318 1898.17L975.617 1895.45L979.115 1891.92L971.443 1891.92L971.443 1888.0700000000002L979.103 1888.0700000000002L975.617 1884.5500000000002L978.318 1881.8300000000002Z " transform="matrix(1,0,0,1,-971,-1881)"></path></symbol></svg></a>
    </span>
  </div>
</div>
```
4. Go to the decorative link:  http://localhost:3000/?p=atoms-decorative-link
5. Check its markup. Find no hidden text and its container element `<span>`, when it doesn't have value.  Also, 2 attributes in the `<a>`, **title** and **aria-describedby** are not added when they don't have their values.
```
<span class="ma__decorative-link">
  <a href="#" class="js-clickable-link">
    Lorem ipsum dolor sit amet.
    &nbsp;<svg aria-hidden="true"><use xlink:href="#27404a1a788cb213d119ddcafde3834c.3"></use></svg><svg xmlns="http://www.w3.org/2000/svg" style="display: none"><symbol xmlns="http://www.w3.org/2000/svg" aria-hidden="true" version="1.1" viewBox="0 0 16 18" id="27404a1a788cb213d119ddcafde3834c.3"><path d="M983.721 1887.28L983.721 1887.28L986.423 1890L986.423 1890L986.423 1890L983.721 1892.72L983.721 1892.72L978.318 1898.17L975.617 1895.45L979.115 1891.92L971.443 1891.92L971.443 1888.0700000000002L979.103 1888.0700000000002L975.617 1884.5500000000002L978.318 1881.8300000000002Z " transform="matrix(1,0,0,1,-971,-1881)"></path></symbol></svg>
  </a>
</span>
```
## Screenshots
<img width="521" alt="pattern_lab_-_pages-service" src="https://user-images.githubusercontent.com/9633303/51752876-7c9ed000-2086-11e9-8bae-1d3cef365ad8.png">


## Additional Notes:

Anything else to add?

#### Impacted Areas in Application


* 

#### @TODO


*

#### Today I learned...
